cloudfoundry/resource_cf_user_provided_service.go

```diff
@@ -220,8 +225,9 @@ func resourceUserProvidedServiceDelete(d *schema.ResourceData, meta interface{})
 	session.Log.DebugMessage("begin resourceServiceInstanceDelete")
 
 	sm := session.ServiceManager()
+	recursiveDelete := d.Get("recursive_delete").(bool)
 
-	if err = sm.DeleteServiceInstance(d.Id()); err != nil {
+	if err = sm.DeleteServiceInstance(d.Id(), recursiveDelete); err != nil {
```

Comment:
this should be calling sm.DeleteUserProvidedService() instead 

https://github.com/mevansam/terraform-provider-cf/blob/98f04b8016e18301f3fbee1ea5f922e870cb2feb/cloudfoundry/cfapi/service_manager.go#L633-L637

Note that there is no associated recursive delete for the UPS according to http://apidocs.cloudfoundry.org/4.4.0/user_provided_service_instances/delete_a_particular_user_provided_service_instance.html